lib/sycamore/sycamore/data/table.py

```diff
@@ -112,6 +114,31 @@ def __init__(self, cells: list[TableCell], caption: Optional[str] = None):
         self.num_rows = max(max(c.rows) for c in self.cells) + 1
         self.num_cols = max(max(c.cols) for c in self.cells) + 1
 
+        if column_headers is not None:
+            self.column_headers = column_headers
+        else:
+            header_rows = sorted(set((row_num for cell in self.cells for row_num in cell.rows if cell.is_header)))
+
+            max_header_prefix_row = -1
+            for i in range(len(header_rows)):
+                if header_rows[i] != i:
+                    break
+                max_header_prefix_row = i
+
+            header_array = np.empty([max_header_prefix_row + 1, self.num_cols], dtype="object")
+            if len(self.cells) > 0:
+                for cell in self.cells:
+                    if cell.is_header and cell.rows[0] <= max_header_prefix_row:
+                        for col in cell.cols:
+                            header_array[cell.rows[0], col] = cell.content
+                        for row in cell.rows[1:]:
+                            for col in cell.cols:
+                                header_array[row, col] = ""
+
+            self.column_headers = []
+            for npcol in header_array.transpose():
+                self.column_headers.append(" | ".join(OrderedDict.fromkeys((c for c in npcol if c not in [None, ""]))))
```

Comment:
any chance we can factor this as a function in common with the pandafication function too?

Reply:
done